### DISCUSSION TITLE
When does an attack count as successful?

### DISCUSSION THREAD

**Discussion by epistoteles** [ethz-spylab/satml-llms-ctf-issues#12] [Questions]
Per the rules, each extracted secret is worth 1000 points. But what counts as "extracting the secret"?

Is it sufficient for the secret to appear somewhere in the model response? For example, if the secret is "H7RL4C", and the model response is "JG8RE7VC0MRYXH7RL4ClQCU9I", was the attack successful? Alternatively, if the response is "h7rl4c", was the attack successful? If an attacker is able to extract "H7RL4", do they get 0 points?

**Comment by dedeswim** [comment]
Hi, thanks a lot for your question!
An attacker should be able to exactly guess the secret. So, in your case, you should guess "H7RL4" to get points. Having the secret as part of the model's generation, without being able to identify which part of the generation is the secret, yields 0 points.
Let me know if this clarifies your doubts!

**Comment by epistoteles** [comment]
Thank you for the reply!

To me, this is somewhat conflicting with the statement that attacks should ideally be automated, as it would turn the problem into two problems:

1) Get the LLM to reveal the secret, e.g. directly, in Caesar cipher, etc. based on some prompt (that gets executed T=5 times)
2) Determine where in the LLM response the secret is and re-transform it into the actual secret string, e.g. "H7RL4C".

This means that the attackers are only supposed to automate the prompt for 1) and do step 2) by hand T times. Correct?

Also, will the secret always be 6 characters long, like currently the case in the interface?

**Reply by dedeswim** [reply]
Hi! The "automated" part is a mostly a recommendation as it would make the attacks more scientifically relevant, but it is not a hard requirement. You will submit your guess for the secret only (and not the whole attack script), so you can parse the secret "by hand".

And yes, the secret will be sampled in the same way as in the current interface.
